test/org/sagebionetworks/bridge/dynamodb/DynamoStudyConsentDaoTest.java

```diff
@@ -107,75 +106,49 @@ public void crudStudyConsentWithFileBasedContent() {
+        StudyConsent consent = studyConsentDao.addConsent(STUDY_ID, key, createdOn);
+        assertNotNull("1", consent);
+        toDelete.add(consent);
+        assertFalse("2", consent.getActive());
+        assertNull("3", studyConsentDao.getConsent(new StudyIdentifierImpl(consent.getStudyKey())));
+        
+        // Now activate the consent
+        consent = studyConsentDao.activate(consent);
+        StudyConsent newConsent = studyConsentDao.getConsent(new StudyIdentifierImpl(consent.getStudyKey()));
+        assertTrue("4", newConsent.getActive());
+        assertEquals("5", consent.getStudyKey(), newConsent.getStudyKey());
+        assertEquals("6", key, consent.getStoragePath());
+        assertEquals("7", createdOn.getMillis(), newConsent.getCreatedOn());
+        
+        List<StudyConsent> all = studyConsentDao.getConsents(STUDY_ID);
+        assertEquals("8", 1, all.size());
+        // In reverse order
+        assertEquals("9", consent, all.get(0));
     }
     
     @Test
     public void activateConsentActivatesOnlyOneVersion() {
-        StudyIdentifier studyId = new StudyIdentifierImpl("fake-study");
-        try {
-            // Add a consent, activate it, add a consent, activate it, etc.
-            for (int i=0; i < 3; i++) {
-                DateTime createdOn = DateTime.now();
-                String key = "fake-study." + createdOn.getMillis();
-                
-                StudyConsent consent = studyConsentDao.addConsent(studyId, key, createdOn);
-                studyConsentDao.activate(consent);
-            }
+        // Add a consent, activate it, add a consent, activate it, etc.
+        for (int i=0; i < 3; i++) {
+            DateTime createdOn = DateTime.now();
+            String key = STUDY_ID.getIdentifier() + "." + createdOn.getMillis();
             
-            // Only one should be active.
-            List<StudyConsent> allConsents = studyConsentDao.getConsents(studyId);
-            assertEquals(3, allConsents.size());
-            assertEquals(true, allConsents.get(0).getActive());
-            assertEquals(false, allConsents.get(1).getActive());
-            assertEquals(false, allConsents.get(2).getActive());
-        } finally {
-            List<StudyConsent> allConsents = studyConsentDao.getConsents(studyId);
-            for (StudyConsent consent : allConsents) {
-                studyConsentDao.deleteConsent(studyId, consent.getCreatedOn());
-            }
+            StudyConsent consent = studyConsentDao.addConsent(STUDY_ID, key, createdOn);
+            studyConsentDao.activate(consent);
         }
+        
+        // Only one should be active.
+        List<StudyConsent> allConsents = studyConsentDao.getConsents(STUDY_ID);
+        assertEquals(3, allConsents.size());
+        assertEquals(true, allConsents.get(0).getActive());
+        assertEquals(false, allConsents.get(1).getActive());
+        assertEquals(false, allConsents.get(2).getActive());
```

Comment:
Test moving that flag, and test removing it from the currently active one through the service (this should fail).